crates/compiler/solve/tests/solve_expr.rs

```diff
@@ -522,6 +522,41 @@ mod solve_expr {
         );
     }
 
+    #[test]
+    fn my_bug() {
+        infer_queries!(
+            indoc!(
+                r#"
+Parser : [
+    Specialize Parser,
+    Record (List {parser: Parser}),
+]
+
+printCombinatorParser : Parser -> Str
+printCombinatorParser = \parser ->
+    when parser is
+#        ^^^^^^
+        Specialize p ->
+            printed = printCombinatorParser p
+            if Bool.false then printed else "foo"
+        Record fields ->
+            fields
+                |> List.map \f ->
+                    printed = printCombinatorParser f.parser
+                    if Bool.false then printed else "foo"
+                |> List.first
+                |> Result.withDefault ("foo")
+
+printCombinatorParser (Record [])
+    "#
```

Comment:
```suggestion
                r#"
                Parser : [
                    Specialize Parser,
                    Record (List {parser: Parser}),
                ]
                
                printCombinatorParser : Parser -> Str
                printCombinatorParser = \parser ->
                    when parser is
                #        ^^^^^^
                        Specialize p ->
                            printed = printCombinatorParser p
                            if Bool.false then printed else "foo"
                        Record fields ->
                            fields
                                |> List.map \f ->
                                    printed = printCombinatorParser f.parser
                                    if Bool.false then printed else "foo"
                                |> List.first
                                |> Result.withDefault ("foo")
                
                printCombinatorParser (Record [])
                "#
```

to line up with how other tests of this type are indented in this file